pkg/controller/proxyconfig/status.go

```diff
@@ -0,0 +1,126 @@
+package proxyconfig
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// syncProxyStatus computes the current status of proxy and
+// updates status if any changes since last sync.
+func (r *ReconcileProxyConfig) syncProxyStatus(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network, cluster *corev1.ConfigMap) error {
+	var err error
+	var noProxy string
+	updated := proxy.DeepCopy()
+
+	if isSpecNoProxySet(&proxy.Spec) {
+		if proxy.Spec.NoProxy == noProxyWildcard {
+			noProxy = proxy.Spec.NoProxy
+		} else {
+			noProxy, err = mergeUserSystemNoProxy(proxy, infra, network, cluster)
+			if err != nil {
+				return fmt.Errorf("failed to merge user/system noProxy settings: %v", err)
+			}
+		}
+	}
+
+	updated.Status.HTTPProxy = proxy.Spec.HTTPProxy
+	updated.Status.HTTPSProxy = proxy.Spec.HTTPSProxy
+	updated.Status.NoProxy = noProxy
+
+	if !proxyStatusesEqual(proxy.Status, updated.Status) {
+		if err := r.client.Status().Update(context.TODO(), updated); err != nil {
+			return fmt.Errorf("failed to update proxy status: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// mergeUserSystemNoProxy merges user-supplied noProxy settings from proxy
+// with cluster-wide noProxy settings, returning a merged, comma-separated
+// string of noProxy settings.
+func mergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network, cluster *corev1.ConfigMap) (string, error) {
+	apiServerURL, err := url.Parse(infra.Status.APIServerURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse API server URL")
+	}
+
+	internalAPIServer, err := url.Parse(infra.Status.APIServerInternalURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse API server internal URL")
+	}
+
+	set := sets.NewString(
+		"127.0.0.1",
+		"localhost",
+		network.Status.ServiceNetwork[0],
+		apiServerURL.Hostname(),
+		internalAPIServer.Hostname(),
+	)
+
+	// TODO: This will be flexible when master machine management is more dynamic.
+	type installConfig struct {
+		ControlPlane struct {
+			Replicas string `json:"replicas"`
+		} `json:"controlPlane"`
+		Networking struct {
+			MachineCIDR string `json:"machineCIDR"`
+		} `json:"networking"`
+	}
+	var ic installConfig
+	data, ok := cluster.Data["install-config"]
+	if !ok {
+		return "", fmt.Errorf("missing install-config in configmap")
+	}
+	if err := yaml.Unmarshal([]byte(data), &ic); err != nil {
+		return "", fmt.Errorf("invalid install-config: %v\njson:\n%s", err, data)
+	}
+
+	switch infra.Status.PlatformStatus.Type {
+	case configv1.AWSPlatformType:
+		set.Insert("169.254.169.254", ic.Networking.MachineCIDR)
+	default:
+		return "", fmt.Errorf("unsupported infrastructure provider: %s", infra.Status.PlatformStatus.Type)
```

Comment:
I'm still a bit confused, then - won't this cause the network operator to be Degraded for all platforms except AWS?

Reply:
Yes, but I have a PR lined up to add GCP and Azure.